akka-actor-typed-tests/src/test/scala/akka/actor/typed/TimerSpec.scala

```diff
@@ -308,8 +308,16 @@ class TimerSpec extends ScalaTestWithActorTestKit(
       val probe = TestProbe[Array[StackTraceElement]]()
       spawn(next(0, probe.ref))
       val elements = probe.expectMessageType[Array[StackTraceElement]]
-      if (elements.count(_.getClassName == "TimerInterceptor") > 1)
+      if (elements.count(_.getClassName contains "TimerInterceptor") > 1)
         fail(s"Stack contains TimerInterceptor more than once: \n${elements.mkString("\n\t")}")
     }
+
+    "typecheck when using withKeyType" in {
+      Behaviors.withTimers[String] { timers0 ⇒
+        val timers = timers0.withKeyType[String]
```

Comment:
was there a mistake you made with the old API that motivated this change? I'm not sure I see the value but if others do then happy to merge

Reply:
Personally, I haven't made this mistake yet.  But there's an easy risk of using the wrong key type, breaking your program.  For instance, if you have a reasonably complex actor, where you're passing the timers instance around various behaviour-defining methods, like one of the ones in the examples of timers.

Reply:
I'm not sure it's worth the additional API complexity. If we think it's important we should enforce specifying the key type in `withTimers` rather than having a default Any and having to opt in to something safer (probably only found after doing the mistake)

Reply:
The downside of that is you lose some nice type inference when wrapping, because it needs you to define the type of the key, and therefore you also have to define the type of message... 😕

So you go from

```scala
Behaviors.withTimers { timers =>
  Behaviors.same[Message]
}
```

to

```scala
Behaviors.withTimers[String, Message] { timers =>
  Behaviors.same[Message]
}
```

Or we could do some of those shenanigans that allow for "currying type parameters"...